Backpack/Alert/Classes/BPKAlertView.m

```diff
@@ -0,0 +1,235 @@
+#import <Backpack/Shadow.h>
+
+
+@interface BPKAlertView()
+
+@end
+
+@implementation BPKAlertView
+{
+    UIView *_circularView;
+    UIView *_iconContainerView;
+    UIImageView *_iconImageView;
+    
+    UIView *_contentHolderView;
+    BPKLabel *_titleLabel;
+    BPKLabel *_descriptionLabel;
+    NSArray<BPKButton *> *_buttons;
+    UIStackView *_buttonStackView;
+    
+    NSDictionary<BPKAlertButtonConfiguration *, BPKButton *> *_buttonConfigurationMap;
+}
+
+- (instancetype)initWithFrame:(CGRect)frame
+{
+    self = [super initWithFrame:frame];
+    if (self) {
+        [self setupViews];
+        [self addViews];
+        [self setupConstraints];
+        
+        
+        UITapGestureRecognizer *tap = [[UITapGestureRecognizer alloc] initWithTarget:self action:@selector(tapView:)];
+        tap.delegate = self;
+        [self setUserInteractionEnabled:YES];
+        [self addGestureRecognizer:tap];
+    }
+    return self;
+}
+
+- (void)setupViews {
+    _contentHolderView = [UIView new];
+    _contentHolderView.backgroundColor = BPKColor.white;
+    _contentHolderView.layer.cornerRadius = 4;
+    
+    _circularView = [[UIView alloc] initWithFrame:CGRectZero];
+    _circularView.layer.cornerRadius = 35;
+    _circularView.backgroundColor = BPKColor.white;
+    
+    _iconContainerView = [[UIView alloc] initWithFrame:CGRectZero];
+    _iconContainerView.layer.cornerRadius = 30;
+    _iconContainerView.backgroundColor = BPKColor.red500;
+    
+    _iconImageView = [[UIImageView alloc] init];
+    
+    _titleLabel = [[BPKLabel alloc] initWithFontStyle:BPKFontStyleTextXlEmphasized];
+    
+    _descriptionLabel = [[BPKLabel alloc] initWithFontStyle:BPKFontStyleTextLg];
+    _descriptionLabel.numberOfLines = 0;
+    _descriptionLabel.textAlignment = NSTextAlignmentCenter;
+    
+    _buttonStackView = [[UIStackView alloc] initWithFrame:CGRectZero];
+    _buttonStackView.axis = UILayoutConstraintAxisVertical;
+    _buttonStackView.spacing = BPKSpacingMd;
+    
+    _iconImageView.translatesAutoresizingMaskIntoConstraints = NO;
+    _buttonStackView.translatesAutoresizingMaskIntoConstraints = NO;
+    _titleLabel.translatesAutoresizingMaskIntoConstraints = NO;
+    _descriptionLabel.translatesAutoresizingMaskIntoConstraints = NO;
+    _circularView.translatesAutoresizingMaskIntoConstraints = NO;
+    _iconContainerView.translatesAutoresizingMaskIntoConstraints = NO;
+    _contentHolderView.translatesAutoresizingMaskIntoConstraints = NO;
+}
+
+- (void)addViews {
+    [self addSubview:_contentHolderView];
+    [self addSubview:_circularView];
+    [_circularView addSubview:_iconContainerView];
+    [_circularView addSubview:_iconImageView];
+
+    [_contentHolderView addSubview:_titleLabel];
+    [_contentHolderView addSubview:_descriptionLabel];
+    [_contentHolderView addSubview:_buttonStackView];
+}
+
+- (void)setupConstraints {
+    [_circularView.topAnchor constraintEqualToAnchor:_circularView.superview.topAnchor].active = YES;
```

Comment:
`+ (void)activateConstraints:(NSArray<NSLayoutConstraint *> *)constraints` is very helpful for methods like this

Reply:
👍